enhancements/olm-deprecate-copied-csvs.md

```diff
@@ -0,0 +1,207 @@
+---
+title: deprecate-copied-csvs
+authors:
+  - "@njhale"
+reviewers:
+  - TBD
+  - "@"
+approvers:
+  - TBD
+  - "@kevinrizza"
+creation-date: 2021-09-27
+last-updated: 2021-10-25
+status: implementable
+see-also:
+  - "/enhancements/olm-deprecate-copied-csvs.md"
+---
+
+# Deprecate Copied CSVs 
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [olm-docs](https://github.com/operator-framework/olm-docs/)
+
+## Summary
+
+This enhancement details the steps required to deprecate and replace OLM's [Copied CSV](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/#copied-csvs) feature in higher fidelity than [ever before](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#deprecate-copied-clusterserviceversions). Importantly, it introduces a feature toggle that disables/enables CSV copying for all of OLM.
+
+## Motivation
+
+When an Operator is installed by OLM, a stripped down copy -- e.g. a stub -- of its [CSV](https://olm.operatorframework.io/docs/concepts/crds/clusterserviceversion/) is created in every namespace the Operator is configured to watch (see [OperatorGroup](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/); collectively, these are known as _Copied CSVs_. This means that the number of Copied CSVs on a given cluster grows proportionally with both the number of namespaces and installed Operators. In the worst case, a cluster can have a Copied CSV for every Operator in every namespace, that is `|copied| = |operators|*|namespaces|`. For especially large clusters, with namespaces and installed operators tending in the hundreds or thousands, Copied CSVs consume an untenable amount of resources; e.g. OLM's memory usage, cluster Etcd limits, networking, etc. For these reasons, it's imperative that Copied CSVs be replaced with a more resource friendly successor.
+
+### Goals
```

Comment:
In the context of downstream - does this proposal need to be backportable?

Reply:
Should we add that as a goal then?

Reply:
good idea. added.

Reply:
done